ctapipe/image/extractor.py

```diff
@@ -1285,3 +1290,373 @@ def __call__(
+
+    sum = float64(0.0)
+    jsum = float64(0.0)
+    j = peak_index
+    while j >= 0 and waveforms[j] > descend_limit:
+        sum += waveforms[j]
+        jsum += j * waveforms[j]
+        j -= 1
+        if waveforms[j] > peak_amplitude:
+            descend_limit = rel_descend_limit * peak_amplitude
+
+    j = peak_index + 1
+    while j < nsamples and waveforms[j] > descend_limit:
+        sum += waveforms[j]
+        jsum += j * waveforms[j]
+        j += 1
+        if waveforms[j] > peak_amplitude:
+            descend_limit = rel_descend_limit * peak_amplitude
+
+    if sum != 0.0:
+        cog[0] = jsum / sum
+
+
+class FlashCamExtractor(ImageExtractor):
+    upsampling = IntTelescopeParameter(
+        default_value=4, help="Define the upsampling factor for waveforms"
+    ).tag(config=True, min=1)
+
+    window_width = IntTelescopeParameter(
+        default_value=7, help="Define the width of the integration window"
+    ).tag(config=True, min=1)
+
+    window_shift = IntTelescopeParameter(
+        default_value=3,
+        help="Define the shift of the integration window from the peak_index "
+        "(peak_index - shift)",
+    ).tag(config=True)
+
+    apply_integration_correction = BoolTelescopeParameter(
+        default_value=True, help="Apply the integration window correction"
+    ).tag(config=True)
+
+    local_weight = IntTelescopeParameter(
+        default_value=0,
+        help="Weight of the local pixel (0: peak from neighbors only, "
+        "1: local pixel counts as much as any neighbor)",
+    ).tag(config=True)
+
+    effective_time_profile_std = FloatTelescopeParameter(
+        default_value=2.0,
+        help="Effective Cherenkov time profile std. dev. (in nanoseconds) to "
+        "assume for calculating the gain correction",
+    ).tag(config=True, min=0)
+
+    neighbour_sum_clipping = FloatTelescopeParameter(
+        default_value=5.0,
+        help="(Soft) clipping level of a pixel's contribution to a neighbour sum "
+        "(set to 0 or inf to disable clipping)",
+    ).tag(config=True)
+
+    leading_edge_timing = BoolTelescopeParameter(
+        default_value=True, help="Calculate leading edge time instead of peak time"
+    ).tag(config=False)
+
+    leading_edge_rel_descend_limit = FloatTelescopeParameter(
+        default_value=0.05,
+        help="Fraction of the peak value down to which samples are accumulated "
+        "in the leading edge centroid calculation",
+    ).tag(config=True, min=0.0, max=1.0)
+
+    def __init__(self, subarray, **kwargs):
+        super().__init__(subarray=subarray, **kwargs)
+
+        self.sampling_rate_ghz = {
+            tel_id: telescope.camera.readout.sampling_rate.to_value("GHz")
+            for tel_id, telescope in subarray.tel.items()
+        }
+
+        def time_profile_pdf_gen(std_dev: float):
+            if std_dev == 0:
+                return None
+            return scipy.stats.norm(0.0, std_dev).pdf
+
+        self.deconvolution_pars = {
```

Comment:
If you store the deconvolution parameters here, is the `lru_cache` still needed?